api/src/pcapi/utils/db.py

```diff
@@ -73,20 +74,28 @@ def __init__(self, enum_class):  # type: ignore [no-untyped-def] # pylint: disab
     process_literal_param = sqla_types.TypeDecorator.process_literal_param
 
     @property
-    def python_type(self):  # type: ignore [no-untyped-def]
+    def python_type(self) -> typing.Type[enum.Enum]:
         return self.enum
 
-    def copy(self):  # type: ignore [no-untyped-def]
+    def copy(self, **kwargs: typing.Any) -> "MagicEnum":
```

Comment:
Les `kwargs` devraient être typables comme un `dict` plutôt que `Any` non ?

Reply:
De ce que j'ai compris, les types de `args` et `kwargs` doivent être les types de chaque argument. Par exemple, si on sait que tous les arguments passés en `args` sont des entiers, on peut typer `args: int` (pas `tuple[int]`). Idem pour `kwargs: int` (pas `dict[str, int]`). Sinon, il faut choisir `Any`. Sources : 

- https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values
- https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html